### DISCUSSION TITLE
ContractFunctionExecutionError: An unknown error occurred while executing the contract function

### DISCUSSION THREAD

**Discussion by wmahmood1984** [wevm/wagmi#2361] [Q&A]
Hi, 

my wagmi integration works fine in development environment. however, in prod environment, it is giving me this error 
ContractFunctionExecutionError: An unknown error occurred while executing the contract function

and also no write function work. 

Please help me

**Comment by 0xfulgore** [comment]
I think the latest version has broken the way things used to work, and the documentation has not been updated to reflect how to use it.

I noticed in testing that every write function would try and utilise 'homestead' and the publicClient which ofcourse does not ever work and ends in the unknown error.

What I have done is make use of 'walletClient' which I believe is the new way to interact with write functions on a clients wallets behalf; I did not attempt to get this working with hooks and just changed my code to call writeContract direct. 
`import { createPublicClient, createWalletClient, http } from 'viem'`

`const walletClient = createWalletClient({
    chain: mainnet,
    transport: custom(window.ethereum)
  })`

`function mintWrite() {
        try {
            var mintConfig = {
                address: 'xxxxxxx',
                abi: abiJSON,
                functionName: 'mint',
                args: [1],
                gas: 80000
            };
            walletClient.writeContract(mintConfig);
        } catch {
            alert("Error processing");
        }
    }`

If anyone has a better way to do this, chime in, but this worked for me until the documentation has been updated.

**Reply by jxom** [reply]
`writeContract` utilises the Wallet Client already. Do you have a minimal reproducible example you can provide so I can have a look into the error you are getting?

**Reply by wmahmood1984** [reply]
The problem is that i am not able to read the contract and am getting above error. The writing contract is also not working. It seems that there is some problem is config or the project key which is not allowing to read / write in production environment. 

can you please guide what can be the possible reason for this? 

Regards

**Reply by Fataydo** [reply]
Did you find a fix for this? I have this problem too, out of nowhere.

**Reply by 0xfulgore** [reply]
Not entirely I just used the wallet client solution above that I came across. It's at the bottom of my original post

**Comment by 0xfulgore** [comment]
@jxom example https://gist.github.com/abstractdrift/4abfd76701cf25001ddf746b2ee3c790

When using writeContract it seems to default to the publicClient and bail out with a execution error; this is pretty much copy paste of examples on the wagmi docs. Did not matter if using the react hooks or direct calls, both would fail.

**Comment by LanreAkintayo** [comment]
> Hi,
> 
> my wagmi integration works fine in development environment. however, in prod environment, it is giving me this error ContractFunctionExecutionError: An unknown error occurred while executing the contract function
> 
> and also no write function work.
> 
> Please help me

Have you been able to fix it? I'm encountering the same error.

**Reply by wmahmood1984** [reply]
hi, 

Had to change the wagmi version and it worked

**Reply by LuiJunYe** [reply]
can provide sample example on how it works as refer to offical documentation even upgrade wagmi version to 1.3.10 also got the execution error
